### PR TITLE
test(generator): rename test service and proto to MediaService and add GetMediaMetadata method

### PR DESCRIPTION
**Pull request by whowes**
Previously we'd checked in a copy of `resumable_upload.proto` from showcase to be used in generator unit tests for resumable uploads. This was confusing because

- there's only one method listed in the proto, which revealed some generator edge cases e.g. for test class generation
- the name triggered awkward class names like "HttpJsonResumableUploadServiceResumableUploadStub"

This PR changes the test proto to `media.proto`, the service to `MediaService`, and adds a `GetMediaMetadata` method, with the intention to make diffs/testing for future changes more understandable.

### CODE REVIEW

File: sdk-platform-java/gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java [hunk 1]
```diff
@@ -139,6 +139,7 @@ enum SelectiveGapicType {
   static final ImmutableList<Pattern> RESUMABLE_UPLOAD_ALLOWLIST_PATTERNS =
       ImmutableList.of(
           Pattern.compile("^google\\.showcase\\.v1beta1\\.ResumableUploadService\\.UploadMedia$"),
+          Pattern.compile("^google\\.showcase\\.v1beta1\\.MediaService\\.UploadMedia$"),
```
Comment (blakeli0) [inline sdk-platform-java/gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java:143]: nit: Add a comment that this is for unit tests only.

**Review by blakeli0 — APPROVED** [review]

File: sdk-platform-java/gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java (hunk 1 above)
Comment (whowes) [reply sdk-platform-java/gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java:143]: Done.
